plutus-tx-plugin/test/Budget/Spec.hs

```diff
@@ -461,6 +480,20 @@ compiledNull = $$(compile [||
       let ls = [1,2,3,4,5,6,7,8,9,10] :: [Integer]
        in PlutusTx.null ls ||])
 
+compiledListIndexing :: CompiledCode ([PlutusTx.BuiltinData] -> PlutusTx.BuiltinData)
+compiledListIndexing = $$(compile [||
+      \xs -> xs List.!! 5 ||])
+
+compiledBuiltinListIndexing :: CompiledCode (PlutusTx.BuiltinData -> PlutusTx.BuiltinData)
+compiledBuiltinListIndexing = $$(compile [||
+      \d -> BI.unsafeDataAsList d `List.indexBuiltinList` 5 ||])
```

Comment:
why are you taking a `Data` argument rather than a builtin list directly?

Reply:
I didn't manage to lift a `BuiltinList`. Either the required instance is missing or I'm not doing it right.

Reply:
It exists:

```haskell
-- See Note [Lift and Typeable instances for builtins]
instance (FromBuiltin arep a, uni `PLC.HasTermLevel` [a]) => Lift uni (BuiltinList arep) where
    lift = liftBuiltin . fromBuiltin
```

but it doesn't work, because the class structure is [bad](https://input-output-rnd.slack.com/archives/G01EE4NQ9U0/p1699859235629579). The constraints say that `BuiltinData` must be in the universe, which it's not, hence things don't work.

Reply:
I'm confused. They say the type inside the list has to be in there, but here we've got a list of integers, which should be fine?

Reply:
Oh, indeed, it's integers, not datas. I don't understand either then.

Reply:
If it's `BuiltinList BuiltinData`:
```
    • No instance for ‘plutus-core-1.25.0.0:Universe.Core.Contains
                         plutus-core-1.25.0.0:PlutusCore.Default.Universe.DefaultUni
                         PlutusTx.BuiltinData’
        arising from a use of ‘liftCodeDef’
    • In the second argument of ‘unsafeApplyCode’, namely
        ‘liftCodeDef builtinListIndexingInput’
```

If it's `BuiltinList BuiltinInteger`, then even `toBuiltin` doesn't work:

```
    • No instance for ‘PlutusTx.Builtins.Class.ToBuiltin
                         [Integer] (BI.BuiltinList Integer)’
        arising from a use of ‘PlutusTx.toBuiltin’
    • In the expression: PlutusTx.toBuiltin [1 :: Integer .. 10]
```

Reply:
> If it's `BuiltinList BuiltinInteger`, then even `toBuiltin` doesn't work:

Right, that's because the instance truly doesn't exist. I've created a ticket: #5924